ikarus/finiteelements/feresulttypes.hh

```diff
@@ -0,0 +1,146 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#include <type_traits>
+
+#include <ikarus/utils/tensorutils.hh>
+
+/**
+ *
+ * \ingroup FEParameterTags
+ * \brief A strongly typed enum class representing the type of the result request
+ */
+
+namespace Ikarus {
+namespace Util {
+
+  struct VectorizeWithVoigt
+  {
+    template <typename Derived>
+    auto operator()(const Eigen::MatrixBase<Derived>& mat) const {
+      return toVoigt(mat.derived(), false);
+    }
+  };
+
+  struct VectorizeGeneric
+  {
+    template <typename Derived>
+    auto operator()(const Eigen::MatrixBase<Derived>& mat) const {
+      return mat.derived().reshaped().eval();
+    }
+  };
+
+  struct MatricizeWithVoigt
+  {
+    template <typename Derived>
+    auto operator()(const Eigen::MatrixBase<Derived>& vec) const {
+      return fromVoigt(vec.derived(), false);
```

Comment:
Can you maybe generalize this, to also alllow for true and false?

Reply:
Yeah the `asVec` function can take a second argument i think